src/metadata-args/MetadataArgsStorage.ts

```diff
@@ -339,21 +339,28 @@ export class MetadataArgsStorage {
     protected filterByTargetAndWithoutDuplicateProperties<
         T extends { target: Function | string; propertyName: string },
     >(array: T[], target: (Function | string) | (Function | string)[]): T[] {
-        const newArray: T[] = []
-        array.forEach((item) => {
-            const sameTarget = Array.isArray(target)
-                ? target.indexOf(item.target) !== -1
-                : item.target === target
-            if (sameTarget) {
-                if (
-                    !newArray.find(
-                        (newItem) => newItem.propertyName === item.propertyName,
-                    )
-                )
-                    newArray.push(item)
+        const uniqueArray: T[] = []
+        const targets = Array.isArray(target) ? [...target].reverse() : [target]
```

Comment:
The targets are added in the order ChildChild < Child < Base. Therefore, we need to reverse it to apply the parent columns first and override them if they were overridden by the child.